### PR TITLE
Blocks-Backend-Events

### LINKED ISSUE #723: [Blocks] Implement other event triggers besides `click` & support event triggers across multiple components
Currently, the only event trigger we support is `click`. 

1. Can we add support additional event triggers? Some of these will be component specific, but something like an "on change" event would be useful for many different components. For example, in my Block below, I would like to run the function any time the input sliders change value

```python
block = gr.Blocks()

with block:
    gr.Markdown("Let's do some kinematics! Choose the speed and angle to see the trajectory.")
    
    with gr.Row():
        speed = gr.inputs.Slider(1, 30, default=25)
        angle = gr.inputs.Slider(0, 90, default=45)
    output = gr.outputs.Image(type="plot")
    btn = gr.Button("Run")
    btn.click(plot, [speed, angle], output)

block.launch()
```

2. It would be nice if I could easily assign a trigger to "if any of these two sliders change" without having to write two separate event trigger lines

3. It would be _really_ nice if I could just say something like "if anything changes in this Block" re-run all of the function(s). I would assume that many of our users want this functionality for their blocks. Could we have a parameter in `gr.Blocks()` like `live=True`. In other words, if you run `gr.Blocks(live=True)`, then you have an automatically responsive Block in which if any of the input components change, all of the functions are re-run and the outputs automatically refresh.

### PR DESCRIPTION
**Pull request by FarukOzderim**
All events are implemented except `Tabs`. `Tabs` is not a component in backend.

Closes #723

### CODE REVIEW

**Comment by pngwn** [comment]
@FarukOzderim I think `Tabs` _should_ be the same as everything else. It isn't any different functionally from rows, columns or carousel. Grid will probably fall under a similar category.

**Comment by abidlabs** [comment]
I agree with @pngwn, tab change could be a useful event to track. @FarukOzderim since `Tab` is a `BlockContext` which is a `Block`, it should also support the `set_event_trigger()` already, right?

My only additional suggestion here is to have more descriptive docstrings for the event triggers of. Instead :

```
"""
Parameters:
    fn: function name
    inputs: List of inputs
    outputs: List of outputs
Returns: None
"""
```

```
"""
Creates a trigger to call a function when this component is clicked
Parameters:
    fn: the function to call
    inputs: the input components whose values should be used as the inputs to the function
    outputs: the output components whose values should be updated based on the function return values
Returns: None
"""
```


Otherwise, LGTM.

File: gradio/blocks.py [hunk 1]
```diff
@@ -19,77 +19,33 @@ def __init__(self):
...
-                "outputs": [block._id for block in outputs],
-            }
-        )
-
-    def change(
-        self, fn: str, inputs: List["Component"], outputs: List["Component"]
+    def set_event_trigger(
+        self,
+        event_name: str,
+        fn: str,
```
Comment (abidlabs) [inline gradio/blocks.py:25]: Type of `fn` should be `Callable`

**Comment by FarukOzderim** [comment]
I wanted to clarify meaning of tabs below, but now it is clarified!
https://github.com/gradio-app/gradio/issues/723#issuecomment-1073978532

**Comment by FarukOzderim** [comment]
Ready to merge!

**Pull request merged**